controllers/usersignup/usersignup_controller_test.go

```diff
@@ -534,17 +565,95 @@ func TestUserSignupFailedMissingNSTemplateTier(t *testing.T) {
 			Status: v1.ConditionFalse,
 			Reason: "UserIsActive",
 		})
-	assert.Equal(t, "approved", userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey])
-	AssertMetricsCounterEquals(t, 1, metrics.UserSignupApprovedTotal) // incremented, even though the provisioning failed due to missing NSTemplateTier
-	AssertMetricsCounterEquals(t, 1, metrics.UserSignupUniqueTotal)   // incremented, even though the provisioning failed due to missing NSTemplateTier
-	AssertThatCountersAndMetrics(t).
-		HaveMasterUserRecordsPerDomain(toolchainv1alpha1.Metric{
-			string(metrics.External): 1,
-		}).
-		HaveUsersPerActivationsAndDomain(toolchainv1alpha1.Metric{
-			"1,external": 1,
-			"1,internal": 1,
+
+	AssertThatCountersAndMetrics(t).HaveMasterUserRecordsPerDomain(toolchainv1alpha1.Metric{
+		string(metrics.External): 1,
+		string(metrics.Internal): 1,
+	})
+}
+
+func TestUserSignupFailedMissingNSTemplateTier(t *testing.T) {
+
+	type variation struct {
+		description string
+		tierName    string
+		config      *toolchainv1alpha1.ToolchainConfig
+	}
+
+	variations := []variation{
+		{
+			description: "default tier",
+			tierName:    "base",
+			config:      commonconfig.NewToolchainConfigObjWithReset(t, testconfig.AutomaticApproval().Enabled(true)),
+		},
+		{
+			description: "non-default tier",
+			tierName:    "nonexistent",
+			config:      commonconfig.NewToolchainConfigObjWithReset(t, testconfig.AutomaticApproval().Enabled(true), testconfig.Tiers().DefaultTier("nonexistent")),
+		},
+	}
+
+	for _, v := range variations {
+		t.Run(fmt.Sprintf("TestUserSignupFailedMissingNSTemplateTier/%s", v.description), func(t *testing.T) {
```

Comment:
```suggestion
		t.Run(v.description, func(t *testing.T) {
```

you don't need to specify again the name of the parent test when running subtests, so I believe that the "variation" alone is enough here